src/libsystemd-network/test-ndisc-ra.c

```diff
@@ -220,6 +222,11 @@ static void test_radv(void) {
         assert_se(sd_radv_set_rdnss(ra, 0, &test_rdnss, 1) >= 0);
         assert_se(sd_radv_set_rdnss(ra, 0, NULL, 0) >= 0);
 
+        assert_se(sd_radv_set_dnssl(ra, 0, NULL) >= 0);
+        assert_se(sd_radv_set_dnssl(ra, 600, NULL) >= 0);
+        assert_se(sd_radv_set_dnssl(ra, 0, (char **)test_dnssl) >= 0);
+        assert_se(sd_radv_set_dnssl(ra, 600, (char **)test_dnssl) >= 0);
```

Comment:
It seems that `ra` should be freed in order to prevent memory leaks. I've started getting the following error recently:
```sh
==17121== HEAP SUMMARY:
==17121==     in use at exit: 1,784 bytes in 10 blocks
==17121==   total heap usage: 62 allocs, 52 frees, 7,673 bytes allocated
==17121==
==17121== 24 bytes in 1 blocks are definitely lost in loss record 4 of 10
==17121==    at 0x4C2FA50: calloc (vg_replace_malloc.c:711)
==17121==    by 0x10D5E0: sd_radv_set_dnssl (sd-radv.c:621)
==17121==    by 0x10A866: test_radv (test-ndisc-ra.c:228)
==17121==    by 0x10B420: main (test-ndisc-ra.c:379)
==17121==
==17121== 24 bytes in 1 blocks are definitely lost in loss record 5 of 10
==17121==    at 0x4C2FA50: calloc (vg_replace_malloc.c:711)
==17121==    by 0x10D5E0: sd_radv_set_dnssl (sd-radv.c:621)
==17121==    by 0x10AF85: test_ra (test-ndisc-ra.c:329)
==17121==    by 0x10B425: main (test-ndisc-ra.c:380)
==17121==
```

Reply:
Looks like ra->dnssl isn't freed on an unref. Just a sec...

Reply:
Updated, see pull request #6858